src/uct/base/uct_iface.h

```diff
@@ -396,6 +396,110 @@ typedef struct uct_iface_local_addr_ns {
     }
 
 
+/**
+ * Call TL registration routine.
+ *
+ * @param [in] _name  TL name
+ */
+#define uct_tl_register(_name) \
+    uct_tl_##_name##_ctor()
+
+
+/**
+ * Call TL unregistration routine.
+ *
+ * @param [in] _name  TL name
+ */
+#define uct_tl_unregister(_name) \
+    uct_tl_##_name##_dtor()
+
+
+/**
+ * Register component and TL
+ *
+ * @param [in] _name  Component and TL name
+ * @param [in] ...    Transport and component registration routines
+ *                    attributes (static)
+ */
+#define UCT_COMP_TL_REGISTER(_name, ...) \
+    UCT_COMPONENT_REGISTER_DECL(_name, __VA_ARGS__); \
+    UCT_TL_REGISTER_DECL(_name, __VA_ARGS__); \
+    static void uct_component_tl_##_name##_ctor(void); \
+    void UCS_F_CTOR uct_##_name##_init(void) { \
+        uct_component_tl_##_name##_ctor(); \
+        uct_component_register(_name); \
+        uct_tl_register(_name); \
+    } \
+    void uct_component_tl_##_name##_ctor(void) \
+
+
+/**
+ * Unregister component and TL
+ *
+ * @param [in] _name  Component and TL name
+ * @param [in] ...    Transport and component registration routines
+ *                    attributes (static)
+ */
+#define UCT_COMP_TL_UNREGISTER(_name, ...) \
+    UCT_COMPONENT_REGISTER_DECL(_name, __VA_ARGS__); \
+    UCT_TL_REGISTER_DECL(_name, __VA_ARGS__); \
+    static void uct_component_tl_##_name##_dtor(void); \
+    void UCS_F_DTOR uct_##_name##_cleanup(void) { \
+        uct_tl_unregister(_name); \
+        uct_component_unregister(_name); \
+        uct_component_tl_##_name##_dtor(); \
+    } \
+    void uct_component_tl_##_name##_dtor(void) \
```

Comment:
i think the trailing \ is not needed here

Reply:
> i think the trailing \ is not needed here

yep, removed